### PR TITLE
Refer to VDAF instead of "PPM scheme"

### PR DESCRIPTION
**Pull request by cjpatton**
The spec vaguely refers to the "concrete PPM scheme" whenever we need to talk about protocol specific parameters. This change replaces each such reference with a reference to VDAF. It also adds the aggregation parameter to the aggregate request, which will be needed for hits.

This is an alternative #154 that aims to be less invasive. Partially addresses #98.

### CODE REVIEW

**Review by chris-wood — APPROVED** [review]
Much cleaner 👍

File: draft-gpew-priv-ppm.md [hunk 1]
```diff
@@ -682,25 +683,24 @@ parameters except its own, the leader sends a POST request to
 ~~~
 struct {
   TaskID task_id;
-  opaque helper_state<0..2^16>;
+  opqaue agg_param<0..2^16-1>;  // VDAF aggregation parameter
```
Comment (tgeoghegan) [inline draft-gpew-priv-ppm.md:686]:
```suggestion
  opaque agg_param<0..2^16-1>;  // VDAF aggregation parameter
```

**Review by tgeoghegan — APPROVED** [review]
LGTM, modulo one typo

File: draft-gpew-priv-ppm.md [hunk 2]
```diff
@@ -194,22 +194,25 @@ The protocol is executed by a large set of clients and a small set of servers.
 We call the servers the *aggregators*. Each client's input to the protocol is a
 set of measurements (e.g., counts of some user behavior). Given the input set of
 measurements `x_1, ..., x_n` held by `n` users, the goal of a *privacy
-preserving measurement (PPM) protocol* is to compute `y = F(x_1, ..., x_n)` for
-some aggregation function `F` while revealing nothing else about the
-measurements.
+preserving measurement (PPM) protocol* is to compute `y = F(p, x_1, ..., x_n)`
+for some aggregation function `F` and aggregation parameter `p` while revealing
+nothing else about the measurements.
```
Comment (ekr) [inline draft-gpew-priv-ppm.md:199]: I think you need to say something more about "p" here. Like, what is it?

And why can't we just bury that in F. I.e., \mathcal{F} is a family of functions parametrized by p and F is a specific instance?

File: draft-gpew-priv-ppm.md [hunk 3]
```diff
@@ -194,22 +194,25 @@ The protocol is executed by a large set of clients and a small set of servers.
...
 
 This protocol is extensible and allows for the addition of new cryptographic
-schemes that compute new functions. The current version supports two schemes:
+schemes that compute new functions. A compatible scheme implements the VDAF
```
Comment (ekr) [inline draft-gpew-priv-ppm.md:202]:
```suggestion
schemes that implement the VDAF
```

File: draft-gpew-priv-ppm.md [hunk 4]
```diff
@@ -282,15 +285,16 @@ Helper:
...
-* The PPM scheme to use. This is to some extent dictated by the previous
-  choices.
-* The minimum "batch size" of reports which can be aggregated.
+* The type of each measurement;
+* The aggregation function to compute (e.g., sum, mean, etc.) and an optional
+  aggregation parameter;
+* The set of aggregators and necessary cryptographic keying material to use;
+* The VDAF to execute, which to some extent is dictated by the previous
+  choices;
+* The minimum "batch size" of reports which can be aggregated; and
```
Comment (ekr) [inline draft-gpew-priv-ppm.md:297]: Can we remove the semicolons here?

File: draft-gpew-priv-ppm.md [hunk 5]
```diff
@@ -318,10 +322,13 @@ data is "valid". In PPM, input validation is complicated by the fact that none
...
-In order to address this problem, each PPM client generates a zero-knowledge
-proof that its report is valid and attaches it to the report. The aggregators
-can then jointly verify this proof prior to incorporating the report in the
-aggregation and reject the report if it cannot be verified. However, they do not
+In order to address this problem, the aggregators engage in a secure,
+multi-party computation, specified by the chosen VDAF, to ensure that consuming
+the report will not corrupt the aggregate result. The input shares
+generated by the client include information used by the aggregators during
+aggregation. For example, `prio3` includes a distributed zero-knowledge proof of
+the input's validity [BBCGGI19]. The aggregators can then jointly verify this
```
Comment (ekr) [inline draft-gpew-priv-ppm.md:330]:
```suggestion
the input's validity [BBCGGI19] which the aggregators can jointly verify
```

File: draft-gpew-priv-ppm.md [hunk 6]
```diff
@@ -318,10 +322,13 @@ data is "valid". In PPM, input validation is complicated by the fact that none
...
+proof and reject the report if it cannot be verified. However, they do not
```
Comment (ekr) [inline draft-gpew-priv-ppm.md:331]:
```suggestion
reject the report if it cannot be verified. However, they do not
```

File: draft-gpew-priv-ppm.md [hunk 7]
```diff
@@ -730,8 +730,8 @@ previous one (See {{anti-replay}}.)
 
 The response is an HTTP 200 OK with a body consisting of the helper's updated
 state and a sequence of *sub-responses*, where each sub-response corresponds to
-the sub-request in the same position in `AggregateReq`. The structure of each
-sub-response is specific to the PPM protocol:
+the sub-request in the same position in `AggregateReq`. Each sub-response
+encodes the timestamp and nonce, as well as a VDAF-specific `message`:
```
Comment (ekr) [inline draft-gpew-priv-ppm.md:734]:
```suggestion
encodes the timestamp, a nonce, and a VDAF-specific `message`:
```

**Review by ekr — APPROVED** [review]
This is generally good.

@cjpatton can you somehow the F(p, ...) thing before you land?

File: draft-gpew-priv-ppm.md (hunk 7 above)
Comment (cjpatton) [reply draft-gpew-priv-ppm.md]: Took this but merged "timestamp" into "nonce".

File: draft-gpew-priv-ppm.md (hunk 4 above)
Comment (cjpatton) [reply draft-gpew-priv-ppm.md]: Replaced with periods.

File: draft-gpew-priv-ppm.md (hunk 2 above)
Comment (cjpatton) [reply draft-gpew-priv-ppm.md:199]: Two reasons not to bury this:
1. `F` is implied by the VDAF in use
2. `p` is a parameter chosen by the collector and distributed to the aggregators.

I agree we should say more about what this is. I've added a paragraph to add a bit more color.

By the way, the canonical example of an aggregation parameter is the set of candidate prefixes for each round of heavy hitters.

Comment (ekr) [reply draft-gpew-priv-ppm.md:199]: I can live with your text here, but I don't really agree with this rationale. This is introductory text and doesn't need to map 1:1 with the VDAF spec.

**Review by ekr — APPROVED** [review]

File: draft-gpew-priv-ppm.md (hunk 2 above)
Comment (cjpatton) [reply draft-gpew-priv-ppm.md:199]: Yeah I agree. Happy to iterate on it later on.

**Comment by cjpatton** [comment]
Squashed.

**Pull request merged**